### ISSUE TITLE
Make guests AI find the First Aid Room more frequently than sitting on a bench

### ISSUE THREAD

**Issue by Wolfscopez** [OpenRCT2/OpenRCT2#1959]
I have the suggestion of potentially making it that guests have a certain proximity to them whether they would consider going to a First Aid Room to cure their sickness rather than immediately try to find a bench nearby.

**Comment by Patrik356b** [comment]
I don't think I have ever seen people use the first aid room. Sounds like a
good idea.

**Comment by Wolfscopez** [comment]
They do, they just rarely do it, in order for them to actually use them, you pretty much have to force them to use them via placing a First Aid room directly outside of an entrance so that they walk right towards the First-Aid room. If not then they rarely ever use them.

**Comment by duncanspumpkin** [comment]
According to the peep logic they use the first aid room if they want to go on a ride and are sick. There are a bunch of other cases that can occur before wanting to go on a ride so first aid is probably a bit too low in the priority order. It wouldn't be hard to change it so that they have a look for a first aid room when sick (and not part of the find ride code).

**Comment by Wolfscopez** [comment]
I personally would like that to be implemented as I really loathe having to have First-Aid rooms pretty much outside any Roller-Coaster or nauseating ride, or having a handyman patrol a footpath making sure there is no vomit.

**Comment by IntelOrca** [comment]
I can't remember if I improved or not already due to peeps not using their park maps correctly.

**Comment by Wolfscopez** [comment]
I think a good idea is to add to the peep logic that guests will try and find a First-Aid Room if it is in a reasonable range, if it's like on the other side of the park then for them to just sit down instead as they'd probably vomit on the way to the First-Aid room.

So sort of to add a max range as to how far away it can be before they just sit down?
I don't know if that's possible, just an idea

**Comment by PFCKrutonium** [comment]
So basically check for a First Aid if it is within X tiles... Seems reasonable.

**Comment by Wolfscopez** [comment]
Exactly, this is to stop guests trying to walk halfway across the park just to get to a First-Aid Room, because then if that happened, they'd just vomit on the way there.

**Comment by Wolfscopez** [comment]
I don't know if you have addressed the issue or not but you also might want to go and make it higher priority for sick peeps to go and seek a first aid room over sitting on a bench if there is one within a few blocks, some sick peeps are just sitting on a bench adjacent to a first aid room

**Comment by duncanspumpkin** [comment]
@Wolfscopez the part about checking a max range is how it works at present (I think). The issue is that its too far down the peep logic and sick peeps do the sitting check earlier on.

**Comment by tellovishous** [comment]
Pardon the bump here, but yeah I agree that peeps ought to be able to use the first aid stations more often, they're seldom used as is even though they could definitely be useful. I would also add that it could be more frequent if they also have maps on them, it's pretty rare that I see peeps "Heading To First Aid Room" in the guest list window.
